### DISCUSSION TITLE
Sorting File/Directory List

### DISCUSSION THREAD

**Discussion by NullDev42** [MusicPlayerDaemon/MPD#2070] [Q&A]
I currently use a web based frontend ([RompR](https://github.com/fatg3erman/RompR)) which among other things, allows the user to view their collection as a "File Browser".  This allows the user to group things together in a custom way by simply changing the underlying storage folder structure.  The one short coming is it is not sorted alphabetically and seems to retain whatever order the files were first entered into the MPD database.

Digging through the project's code, I see that they use the API command "listall" to create this structure.  I know it's possible that RompR can generate code to manage the sort on their end.  However, I'm wondering if one of the existing API calls could possibly return a similar dataset, but sorted and save RompR the bother.

I've tried using the "search {FILTER} [sort {TYPE}] [window {START:END}]" command, but I cannot seem to get the filter syntax correct.  For instance, I've tried `file == *` and `file != ''` to no avail.  The protocol API doesn't expand much on how wild cards are used with filters, but I assume they are available.

Can someone please shed some additional light on the Filter and Sort syntax in this case?  Thank you!

EDIT: I should have mentioned that I'm currently running mpd-0.23.15

**Comment by jcorporation** [comment]
- As far I know: The search response does not include directories and playlists. Hence this it is not usable for this scenario.
- For myMPD: I use the `lsinfo` command and sort the result in the client.
